controllers/deployment.go

```diff
@@ -671,17 +676,12 @@ func getSecurityContext(master *v1alpha1.CDAPMaster, services ServiceGroup) (*v1
 func getReplicas(master *v1alpha1.CDAPMaster, services ServiceGroup) (int32, error) {
 	replicas := int32(0)
 	for _, service := range services {
-		if spec, err := getCDAPScalableServiceSpec(master, service); err != nil {
-			return 0, nil
-		} else if spec != nil {
-			if spec.Replicas == nil {
-				continue
-			}
-			if replicas == 0 {
-				replicas = *spec.Replicas
-			} else if replicas != *spec.Replicas {
-				return 0, fmt.Errorf("value of field Replicas not the same across (%s)", strings.Join(services, ","))
-			}
+		if serviceReplicas, err := getServiceReplicas(master, service); err != nil {
+			return 0, err
+		} else if replicas == 0 {
+			replicas = serviceReplicas
+		} else if replicas != serviceReplicas {
```

Comment:
just curious: why do we require all services have same number of replicas?

Reply:
I think this was a side effect of allowing running multiple CDAP services in the same pod. The Replicas setting in the CDAP CR is per service, hence if multiple services are running in the same pod, they need to have the same replicas setting in the CR.